webui/react/src/utils/routes.ts

```diff
@@ -0,0 +1,11 @@
+export const isFullPath = (url: string): boolean => {
+  return url.startsWith('http');
+};
+
+export const parseUrl = (url: string): URL => {
+  if (!isFullPath(url)) {
+    if (!url.startsWith('/')) url = '/' + url; // TODO assume url is absolute, or we could throw
```

Comment:
suggestion: avoid overwriting/mutating function parameters.
For this particular case it's not as significant because it's just a string, but would be good practice for when the objects are complex.
We should add this eslint rule: https://eslint.org/docs/rules/no-param-reassign

Reply:
sounds good. and the eslint rule sounds good too and if we need to do that for maybe performance reasons we can ignore it case by case